# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1521076338 25200
#      Wed Mar 14 18:12:18 2018 -0700
# Branch stable
# Node ID d664734ce8eb0ca0c30865489d312cd62a2e27e1
# Parent  92aaa85c06b8caa494aa5089cad1402d5889bb98
Use reverse DNS naming scheme for freedesktop application identifier

* etc/icons/org.octave.Octave.appdata.xml: Rename from octave.appdata.xml.in.
Hardcode the name of the corresponding desktop file.
* etc/icons/org.octave.Octave.desktop.in: Rename from octave.desktop.in.

* etc/module.mk (appdatadir, appdata_DATA, desktopdir, desktop_DATA): New
variables.  (VENDOR): Delete no longer needed variable.
(%canon_reldir%_CLEANFILES): Adjust to renamed files.
(%canon_reldir%_EXTRA_DIST): Likewise.
(%reldir%/icons/octave.appdata.xml): Delete rule.
(%reldir%/icons/org.octave.Octave.desktop): Rename rule target.
(install-icons): Delete custom rules to install appdata and desktop files.
(uninstall-icons): Likewise.

* configure.ac: Remove call to OCTAVE_PROG_DESKTOP_FILE_INSTALL.
* acinclude.m4 (OCTAVE_PROG_DESKTOP_FILE_INSTALL): Delete macro.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -209,18 +209,16 @@ OCTAVE_PROG_TAR_REPRODUCIBLE
 ## Programs used when installing Octave.
 AC_PROG_LN_S
 AC_PROG_MKDIR_P
 
 AC_PROG_INSTALL
 INSTALL_SCRIPT="${INSTALL}"
 AC_SUBST(INSTALL_SCRIPT)
 
-OCTAVE_PROG_DESKTOP_FILE_INSTALL
-
 ## Programs used when running Octave.
 OCTAVE_PROG_GHOSTSCRIPT
 OCTAVE_PROG_GNUPLOT
 OCTAVE_PROG_PAGER
 OCTAVE_PROG_PYTHON
 
 ## Programs used to generate icons file formats.
 OCTAVE_PROG_ICOTOOL
diff --git a/etc/icons/octave.appdata.xml.in b/etc/icons/org.octave.Octave.appdata.xml
rename from etc/icons/octave.appdata.xml.in
rename to etc/icons/org.octave.Octave.appdata.xml
--- a/etc/icons/octave.appdata.xml.in
+++ b/etc/icons/org.octave.Octave.appdata.xml
@@ -14,17 +14,17 @@
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.
 
   You should have received a copy of the GNU General Public License
   along with Octave; see the file COPYING.  If not, see
   <https://www.gnu.org/licenses/>.
 -->
 <component type="desktop">
-  <id>%OCTAVE_DESKTOP_FILE%</id>
+  <id>org.octave.Octave.desktop</id>
   <metadata_license>FSFAP</metadata_license>
   <project_license>GPL-3.0+</project_license>
   <name>GNU Octave</name>
   <summary>Interactive programming environment for numerical computations</summary>
 
   <description>
     <p>
       GNU Octave is a high-level interpreted language, primarily intended for
diff --git a/etc/icons/octave.desktop.in b/etc/icons/org.octave.Octave.desktop.in
rename from etc/icons/octave.desktop.in
rename to etc/icons/org.octave.Octave.desktop.in
diff --git a/etc/module.mk b/etc/module.mk
--- a/etc/module.mk
+++ b/etc/module.mk
@@ -27,16 +27,26 @@ etc_fonts_DATA = \
   %reldir%/fonts/FreeSans.otf \
   %reldir%/fonts/FreeSansBold.otf \
   %reldir%/fonts/FreeSansBoldOblique.otf \
   %reldir%/fonts/FreeSansOblique.otf
 
 %canon_reldir%_EXTRA_DIST += \
   $(etc_fonts_DATA)
 
+appdatadir = $(datadir)/metainfo
+
+appdata_DATA = \
+  %reldir%/icons/org.octave.Octave.appdata.xml
+
+desktopdir = $(datadir)/applications
+
+desktop_DATA = \
+  %reldir%/icons/org.octave.Octave.desktop
+
 icon_IMAGE_FILES = \
   %reldir%/icons/octave-logo.svg \
   %reldir%/icons/octave-sombrero.png
 
 icon_PNG_SIZES = \
   512 \
   256 \
   128 \
@@ -53,40 +63,32 @@ WINDOWS_PNG_ICONS = $(filter %-16.png %-
 
 BUILT_ICONS = \
   $(BUILT_PNG_ICONS) \
   %reldir%/icons/octave-logo.ico
 
 %canon_reldir%_EXTRA_DIST += \
   $(BUILT_ICONS) \
   $(icon_IMAGE_FILES) \
-  %reldir%/icons/octave.appdata.xml.in \
-  %reldir%/icons/octave.desktop.in \
-  %reldir%/icons/octave_branding_samples.svg
+  %reldir%/icons/octave_branding_samples.svg \
+  %reldir%/icons/org.octave.Octave.appdata.xml \
+  %reldir%/icons/org.octave.Octave.desktop.in
 
 image_DATA += \
   $(icon_IMAGE_FILES) \
   %reldir%/icons/octave-logo.ico
 
-VENDOR = www.octave.org
-
 DIRSTAMP_FILES += \
   %reldir%/icons/$(octave_dirstamp)
 
 all-local: all-icons
 
-all-icons: %reldir%/icons/octave.appdata.xml %reldir%/icons/octave.desktop $(BUILT_ICONS)
+all-icons: %reldir%/icons/org.octave.Octave.desktop $(BUILT_ICONS)
 
-%reldir%/icons/octave.appdata.xml: %reldir%/icons/octave.appdata.xml.in | %reldir%/icons/$(octave_dirstamp)
-	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(SED) < $< > $@-t \
-	  -e "s|%OCTAVE_DESKTOP_FILE%|${VENDOR}-octave.desktop|" && \
-	mv $@-t $@
-
-%reldir%/icons/octave.desktop: %reldir%/icons/octave.desktop.in | %reldir%/icons/$(octave_dirstamp)
+%reldir%/icons/org.octave.Octave.desktop: %reldir%/icons/org.octave.Octave.desktop.in | %reldir%/icons/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(SED) < $< > $@-t \
 	  -e "s|%OCTAVE_PREFIX%|${prefix}|" && \
 	mv $@-t $@
 
 $(BUILT_PNG_ICONS): %reldir%/icons/octave-logo.svg | %reldir%/icons/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(RSVG_CONVERT) -w $(lastword $(subst -, ,$(patsubst %.png,%,$@))) $< > $@-t && \
@@ -97,47 +99,36 @@ all-icons: %reldir%/icons/octave.appdata
 	$(ICOTOOL) --create --raw  $(WINDOWS_PNG_ICONS) > $@-t && \
 	mv $@-t $@
 
 install-data-local: install-icons
 
 uninstall-local: uninstall-icons
 
 install-icons:
-	-if test -n "$(DESKTOP_FILE_INSTALL)"; then \
-	  $(DESKTOP_FILE_INSTALL) --dir=$(DESTDIR)$(datadir)/applications \
-	    --vendor $(VENDOR) %reldir%/icons/octave.desktop; \
-	fi
 	for f in $(BUILT_PNG_ICONS); do \
 	  size=`echo $$f | $(SED) -n -e "s/.*-\([0-9]\+\)\.png/\1/p"`; \
 	  if test -f $$f; then d=; else d="$(srcdir)/"; fi; \
 	  $(MKDIR_P) $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps; \
 	  $(INSTALL_DATA) "$$d$$f" $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/octave.png; \
 	done
 	$(MKDIR_P) $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps
 	$(INSTALL_DATA) $(srcdir)/%reldir%/icons/octave-logo.svg $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps/octave.svg
-	$(MKDIR_P) $(DESTDIR)$(datadir)/appdata
-	$(INSTALL_DATA) %reldir%/icons/octave.appdata.xml $(DESTDIR)$(datadir)/appdata/$(VENDOR)-octave.appdata.xml
 
 uninstall-icons:
-	if test -n "$(DESKTOP_FILE_INSTALL)"; then \
-	  rm -f $(DESTDIR)$(datadir)/applications/$(VENDOR)-octave.desktop; \
-	fi
 	for f in $(BUILT_PNG_ICONS); do \
 	  size=`echo $$f | $(SED) -n -e "s/.*-\([0-9]\+\)\.png/\1/p"`; \
 	  rm -f $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/octave.png; \
 	done
 	rm -f $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps/octave.svg
-	rm -f $(DESTDIR)$(datadir)/appdata/$(VENDOR)-octave.appdata.xml
 
 EXTRA_DIST += $(%canon_reldir%_EXTRA_DIST)
 
 %canon_reldir%_CLEANFILES += \
-  %reldir%/icons/octave.appdata.xml \
-  %reldir%/icons/octave.desktop
+  %reldir%/icons/org.octave.Octave.desktop
 
 %canon_reldir%_MAINTAINERCLEANFILES += \
   $(BUILT_ICONS)
 
 CLEANFILES += $(%canon_reldir%_CLEANFILES)
 DISTCLEANFILES += $(%canon_reldir%_DISTCLEANFILES)
 MAINTAINERCLEANFILES += $(%canon_reldir%_MAINTAINERCLEANFILES)
 
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2836,23 +2836,16 @@ I didn't find bison, or the version of b
 support all the features that are required, but it's only a problem
 if you need to reconstruct parse.cc, which is the case if you're
 building from VCS sources.
 "
     OCTAVE_CONFIGURE_WARNING([warn_bison])
   fi
 ])
 dnl
-dnl Find desktop-file-install program.
-dnl
-AC_DEFUN([OCTAVE_PROG_DESKTOP_FILE_INSTALL], [
-  AC_CHECK_PROG(DESKTOP_FILE_INSTALL, desktop-file-install, desktop-file-install, [])
-  AC_SUBST(DESKTOP_FILE_INSTALL)
-])
-dnl
 dnl Find find program.
 dnl
 ## Prefer GNU find if found.
 AN_MAKEVAR([FIND],  [OCTAVE_PROG_FIND])
 AN_PROGRAM([gfind], [OCTAVE_PROG_FIND])
 AN_PROGRAM([find],  [OCTAVE_PROG_FIND])
 AC_DEFUN([OCTAVE_PROG_FIND], [
   AC_CHECK_PROGS(FIND, [gfind find])
